### PR TITLE
[Nexus-4593] no autoblock on HTTP 403 Forbidden

### PR DESCRIPTION
**Pull request by nabcos**
https://builds.sonatype.org/job/nexus-oss-its-feature/jdk=java-6x,label=linux/28/

### CODE REVIEW

**Comment by cstamas** [comment]
✚

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4593/Nexus4593NoAutoblockFor403IT.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        assertThat( ProxyMode.valueOf( getStatus().getProxyMode() ), is( ProxyMode.ALLOW ) );
+
+        // 403 for artifact again
+        startErrorServer( 403, true );
+        // download will not fail because we have a local copy cached, but the remote will be hit b/c maxAge is set to 0
+        downloadArtifact( GavUtil.newGav( "nexus4593", "artifact", "1.0.0" ), "target" );
+
+        assertThat( ProxyMode.valueOf( getStatus().getProxyMode() ), is( ProxyMode.ALLOW ) );
+    }
+
```
Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4593/Nexus4593NoAutoblockFor403IT.java:105]: I think there is still the autoUnblock condition missing here...

start a server returning 500
is shall block ( ProxyMode.AUTO_BLOCK )
then switch server to 403
it shall unblock ( ProxyMode.ALLOW )

-1

Comment (nabcos) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4593/Nexus4593NoAutoblockFor403IT.java:105]: see https://github.com/sonatype/nexus/pull/58/files#L5R151 for BLOCKED_AUTO -> ALLOW transition with a 403 http status.

Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4593/Nexus4593NoAutoblockFor403IT.java:105]: Ow, my bad.
+1

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/remote/commonshttpclient/CommonsHttpClientRemoteStorage.java [hunk 2]
```diff
@@ -535,10 +528,13 @@ public class CommonsHttpClientRemoteStorage
             method.releaseConnection();
 
             // HEAD returned error, but not exception, try GET before failing
-            if ( doGet == false && response != HttpStatus.SC_OK )
+            if ( !doGet && response != HttpStatus.SC_OK )
             {
+                // 401 and 403 will throw an exception in executeMethod, no need to check for those here
                 // try with GET unless some known to fail responses are in
-                doGet = ( response != HttpStatus.SC_UNAUTHORIZED ) && ( response != HttpStatus.SC_FORBIDDEN );
+                // doGet = ( response != HttpStatus.SC_UNAUTHORIZED ) && ( response != HttpStatus.SC_FORBIDDEN );
```
Comment (bdemers) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/remote/commonshttpclient/CommonsHttpClientRemoteStorage.java]: dead line can be removed

**Comment by bdemers** [comment]
+1 nice tests

**Comment by cstamas** [comment]
Nice +1, but the 500 handler reminded me to following projects

https://github.com/sonatype/http-testing-harness
https://github.com/cstamas/teaser

Some of those should be started used in ITs

**Pull request merged**